Assets/Scripts/Player/DrunkennessHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Drinkables;
+
+namespace Player
+{
+    public static class DrunkennessHandler
+    {
+        public static void OnDrink(AbstractDrinkable chosenDrink)
+        {
+            chosenDrink.Drink();
+        }
+
+        public static void OnSoberingUp(IEnumerable<AbstractDrinkable> drinksInBody, float timeSinceLastCheck)
+        {
+            var abstractDrinkables = drinksInBody.ToList();
+            
+            for (var i = abstractDrinkables.Count - 1; i >= 0; i--)
+            {
+                if (abstractDrinkables[i].GetDrunkennessTimer() <= 0.0f)
+                {
+                    abstractDrinkables[i].StopDrinkingEffect();
+                    abstractDrinkables.RemoveAt(i);
+                }
+                else
+                {
+                    abstractDrinkables[i].SetDrunkennessTimer(timeSinceLastCheck);
+                }
+
+                // this is to avoid bugs if the player drinks while the sobering function is being called
+                if (i == 0)
+                {
+                    break;
+                }
+            }
+        }
```

Comment:
I was about to accuse you of being a sociopath for starting at the max and working your way down. But i see the break point and it makes sense. 
though i am pretty sure the way the pipeline works you shouldn't ever run into that problem

Reply:
the reason why i go from the end to the beginning is that it's the most elegant way of removing objects in the list while iterating through the loop.

I probably don't need the if at the end, but I was running into issues when the sobering time was only 6 seconds and i was basically stuck in an eternal loop. I'll have to retest with the beers staying in the body for a longer time to see.

Reply:
How could you be stuck in an eternal loop if the loop condition is `i >= 0` ?
The if at the end should not matter, because even if `i` is equal to 0, it's gonna then be equal to `-1` and the loop will exit on it's own

Or maybe, because you are using `var` as the type of `i` it is using a unsigned int ? Maybe you should replace `i` type by a `int` and you won't need the if at the end of the loop ?